assets/shared/helpers/player.js

```diff
@@ -0,0 +1,552 @@
+		},
+
+		/**
+		 * Add an event listener to the player.
+		 *
+		 * @param {HTMLVideoElement} player    The player element.
+		 * @param {string}           eventName Event name (currently only `timeupdate` is supported)
+		 * @param {Function}         callback  Listener callback.
+		 *
+		 * @return {Function} The function to unsubscribe the event.
+		 */
+		on: ( player, eventName, callback ) => {
+			const transformedCallback = ( event ) => {
+				callback( event.target.currentTime );
+			};
+
+			player.addEventListener( eventName, transformedCallback );
+
+			return () => {
+				player.removeEventListener( eventName, transformedCallback );
+			};
+		},
+	},
+	[ VIDEOPRESS_TYPE ]: {
+		/**
+		 * The embed pattern to check if it's the respective type.
+		 */
+		embedPattern: /(videopress|video\.wordpress)\.com\/.+/i,
+
+		/**
+		 * Initialize the player.
+		 *
+		 * @param {HTMLIFrameElement} element The player element.
+		 *
+		 * @return {Promise<HTMLIFrameElement>} The video player through a promise.
+		 */
+		initializePlayer: ( element ) =>
+			new Promise( ( resolve ) => {
+				// eslint-disable-next-line @wordpress/no-global-event-listener -- Not in a React context.
+				window.addEventListener( 'message', ( event ) => {
+					if (
+						event.source !== element.contentWindow ||
+						event.data.event !== 'videopress_durationchange' ||
+						! event.data.durationMs
+					) {
+						return;
+					}
+
+					// Set the duration to a dataset in order to have it available for later.
+					element.dataset.duration =
+						parseInt( event.data.durationMs, 10 ) / 1000;
+
+					resolve( element );
+				} );
+			} ),
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise.
+		 */
+		getDuration: ( player ) => {
+			const duration = player.dataset.duration;
+			if ( ! duration ) {
+				return Promise.reject(
+					new Error( 'Video duration not found' )
+				);
+			}
+
+			return Promise.resolve( parseFloat( player.dataset.duration ) );
+		},
+
+		/**
+		 * Set the video to a current time.
+		 *
+		 * @param {HTMLIFrameElement} player  The player element.
+		 * @param {number}            seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 */
+		setCurrentTime: ( player, seconds ) => {
+			try {
+				player.contentWindow.postMessage(
+					{
+						event: 'videopress_action_set_currenttime',
+						currentTime: seconds,
+					},
+					'*'
+				);
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Play the video.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise} A promise that resolves if the video was played successfully.
+		 */
+		play: ( player ) => {
+			try {
+				player.contentWindow.postMessage(
```

Comment:
Should we wait for the `videpress_playing` [event](https://github.com/Automattic/videopress#events) to come back? 
Same for the other methods like pausing, etc.

Reply:
I'd say to keep like it is for now, and implement it if needed in the future, just to avoid extra complexity. We'd need to attach a new event, and then remove that after returning to avoid a new attached event every `play`.

But let me know if you really consider it important for some reason, then we implement it. 😉

Reply:
Hey! I think if we check it based on the `videpress_playing`, besides making it more complex, it could also not be super trustworthy. Let's say that we call the `play`. So we add the event to listen the `videpress_playing`, and the user or anything else plays the video. It would fire the `videpress_playing`, and let's say that the `postMessage` failed. We'd have a false positive.

IMO, we could accept that it just considers that the postMessage worked to resolve the promise, and if needed, we could improve that in the future. Does it make sense?

Reply:
It makes sense to me. Let's go with what we have and just take this into account if we ever start seeing any inconsistencies or errors (during testing or later).